File: common/Seremis/SoulCraft/api/util/Coordinate2D.java
package Seremis.SoulCraft.api.util;

public class Coordinate2D {

    public double x, y;

    public Coordinate2D() {

    }

    public Coordinate2D(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double getXCoord() {
        return x;
    }

    public double getYCoord() {
        return y;
    }

    public void setCoords(double x, double y) {
        this.x = x;
        this.y = y;
    }

    @Override
    public String toString() {
        return "Coordinate2D[x: " + x + " y: " + y + "]";
    }
}

File: src/main/java/seremis/geninfusion/api/util/Coordinate2D.scala
package seremis.geninfusion.api.util

class Coordinate2D(var x: Double, var y: Double) {

    def getXCoord: Double = x
    def getYCoord: Double = y

    def setXCoord(x: Double) = this.x = x
    def setYCoord(y: Double) = this.y = y

    def setCoord(x: Double, y: Double) {
        this.x = x
        this.y = y
    }

    override def toString: String = {
        "Coordinate2D[x: " + x + " y: " + y + "]"
    }
}
